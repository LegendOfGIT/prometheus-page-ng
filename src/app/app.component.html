<div class="app">
  <header></header>
</div>

<div class="app-content">
  <router-outlet></router-outlet>
</div>
