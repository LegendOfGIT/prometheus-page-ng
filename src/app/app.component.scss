@import 'src/app/variables';

.app {
  font-family: 'Mulish', sans-serif;
}

.app-content {
  background-color: $page-background;
  color: $content-color;
  flex-direction: column;
  top: 0;

  &h1 {
    font-size: 2.0em;
  }
}

.bot__link {
  font-size: 0;
}

.gdpr__dialog {
  background-color: $container-background-color;
}
