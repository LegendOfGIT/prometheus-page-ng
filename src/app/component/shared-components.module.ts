import { NgModule } from '@angular/core';
import { ItemComponent } from './item/item.component';
import { PipesModule } from '../pipes/pipes.module';
import { CommonModule } from '@angular/common';
import { AddToWishlistComponent } from './wishlist/add-to-wishlist.component';
import { BreadcrumbsComponent } from './breadcrumbs/breadcrumbs.component';
import { CategoryTeaserComponent } from './landing-pages/category-teaser.component';
import { LoadingComponent } from './loading/loading.component';
import { ModeratedTeaserComponent } from './landing-pages/moderated-teaser.component';
import {MessagesComponent} from "./messages/messages.component";

@NgModule({
  declarations: [
    AddToWishlistComponent,
    BreadcrumbsComponent,
    CategoryTeaserComponent,
    ModeratedTeaserComponent,
    ItemComponent,
    LoadingComponent,
    MessagesComponent
  ],
  exports: [
    AddToWishlistComponent,
    BreadcrumbsComponent,
    CategoryTeaserComponent,
    ModeratedTeaserComponent,
    ItemComponent,
    LoadingComponent,
    MessagesComponent
  ],
  imports: [
    CommonModule,

    PipesModule
  ]
})
export class SharedComponentsModule { }
