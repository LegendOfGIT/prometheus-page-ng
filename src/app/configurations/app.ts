export class ApplicationConfiguration {
  public static API_BASE: string = '/api';
  public static SERVICE_REQUESTS_BASE: string = 'https://www.wewanna.shop';
}
