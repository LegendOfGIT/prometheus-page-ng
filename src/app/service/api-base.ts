import { Observable } from 'rxjs';
import { shareReplay } from 'rxjs/operators';
import { HttpHeaders } from '@angular/common/http';

export type ReplacementType = { [key: string]: string };

export abstract class ApiBase {

    private static cache: Map<string, Observable<any>> = new Map();

    constructor(protected apiBasePath: string) { }

    /**
     * Returns the modified URL
     */
    protected get(endpoint: string, pathReplacements?: { [key: string]: string }): string {
        const path = this.apiBasePath;
        let end = endpoint;
        if (!!pathReplacements && Object.keys(pathReplacements).length > 0) {
            Object.keys(pathReplacements).forEach(key => {
                const value = pathReplacements[key] || '';
                end = end.replace(new RegExp(`{${key}}`, 'g'), value);
            });
        }

        return `${path}${end}`.split(/[^?=&]+=[&|$]/).join('');
    }

    protected getDefaultContentType() {
        return {
            headers: new HttpHeaders({ 'Content-Type': 'application/json' })
        };
    }

    protected checkCache<T>(url: string, data: Observable<T>): Observable<T> | undefined {

        if (!ApiBase.cache.has(url)) {
            ApiBase.cache.set(url, data.pipe(shareReplay(1)));
        }

        return ApiBase.cache.get(url);

    }

    protected removeCache(url: string) {
        if (ApiBase.cache.has(url)) {
            ApiBase.cache.delete(url);
        }
    }
}
