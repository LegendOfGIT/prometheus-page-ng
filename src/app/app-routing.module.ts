import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';

import { ItemsComponent } from './component/overview/items.component';

const routes: Routes = [
    { path: ':navigationIdLevelA', component: ItemsComponent },
    { path: ':navigationIdLevelA/:navigationIdLevelB', component: ItemsComponent },
    { path: ':navigationIdLevelA/:navigationIdLevelB/:navigationIdLevelC', component: ItemsComponent },
    { path: '', component: ItemsComponent }
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule { }
