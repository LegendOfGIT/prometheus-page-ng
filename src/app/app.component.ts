import { AfterViewInit, Component, ElementRef, OnInit, ViewChild } from '@angular/core';
import { Subject } from 'rxjs';
import { takeUntil } from 'rxjs/operators';

import { UserService } from './service/user.service';
import { SearchProfilesApiService } from './service/search-profiles-api.service';
import { TrackingService } from './service/tracking.service';
import { WishlistItemsApiService } from './service/wishlist-items-api.service';
import { GdprService } from './service/gdpr.service';
import { GdprDecision } from "./model/gdpr-settings";
import {ConsentService} from "./service/consent-service";

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss']
})
export class AppComponent implements AfterViewInit, OnInit  {
  private destroyedService$ = new Subject();

  @ViewChild('gdpr') gdpr: ElementRef | undefined;

  title = 'prometheus-page';

  constructor(
    private userService: UserService,
    private searchProfilesApiService: SearchProfilesApiService,
    private wishlistItemsService: WishlistItemsApiService,
    private gdprService: GdprService,
    private consentService: ConsentService,
    trackingService: TrackingService
  ) {

  }

  ngOnInit(): void {
    this.initialiseSearchProfiles();
    this.initialiseWishlist();

    setInterval(
      () => this.consentService.deleteInformationStoredWithoutConsent(),
      3000);
  }

  ngAfterViewInit() {
    if (GdprDecision.NoDecision !== this.gdprService.getSettings()?.gdprDecision) {
      return;
    }

    setTimeout(
      () => this.gdpr?.nativeElement.click(),
      5000);
  }

  private initialiseSearchProfiles(): void {
    if (this.searchProfilesApiService.items?.length) {
      return;
    }

    this.searchProfilesApiService.getItems(this.userService.activeUser?.id || '')
      .pipe(takeUntil(this.destroyedService$))
      .subscribe((items) => { this.searchProfilesApiService.items = items || []; });
  }

  private initialiseWishlist(): void {
    if (this.wishlistItemsService.items?.length) {
      return;
    }

    this.wishlistItemsService.getItems(this.userService.activeUser?.id || '')
      .pipe(takeUntil(this.destroyedService$))
      .subscribe((items) => { this.wishlistItemsService.items = items || []; });
  }

}
