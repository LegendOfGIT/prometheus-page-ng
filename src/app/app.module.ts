import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { HttpClientModule } from '@angular/common/http';

import { GoogleLoginProvider, SocialAuthServiceConfig } from '@abacritt/angularx-social-login';
import { SocialLoginModule } from '@abacritt/angularx-social-login';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';

import { AddToWishlistComponent } from './component/wishlist/add-to-wishlist.component';
import { ItemsComponent } from './component/overview/items.component';
import { HeaderComponent } from './component/header/header.component';
import { FooterComponent } from './component/footer/footer.component';
import { NavigationComponent } from './component/navigation/navigation.component';
import { SearchProfilesComponent } from './component/profile/search/search-profiles.component';
import { SearchProfileSelectionComponent } from './component/profile/search/search-profile-selection.component';
import { ProfileComponent } from './component/profile/profile.component';
import { WishlistItemsComponent } from './component/wishlist/wishlist-items.component';
import { ImprintComponent } from './component/legal/imprint.component';
import { TranslationPipe } from './pipes/translation.pipe';
import { StartPageComponent } from './component/landing-pages/start-page.component';
import { CategoryTeaserComponent } from './component/landing-pages/category-teaser.component';
import { SingleProductViewComponent } from './component/product-view/single-product-view.component';
import { UrlEncodePipe } from './pipes/web.pipe';
import { GeneralDataProtectionRegulationComponent } from './component/legal/general-data-protection-regulation.component';

@NgModule({
  declarations: [
    AppComponent,

    AddToWishlistComponent,
    CategoryTeaserComponent,
    GeneralDataProtectionRegulationComponent,
    HeaderComponent, FooterComponent,
    ImprintComponent,
    ItemsComponent,
    NavigationComponent,
    ProfileComponent,
    SearchProfilesComponent, SearchProfileSelectionComponent,
    SingleProductViewComponent,
    StartPageComponent,
    WishlistItemsComponent,

    TranslationPipe,
    UrlEncodePipe
  ],
  imports: [
    BrowserModule,
    AppRoutingModule,
    FormsModule,
    HttpClientModule,
    ReactiveFormsModule,
    SocialLoginModule
  ],
  providers: [
    {
      provide: 'SocialAuthServiceConfig',
      useValue: {
        autoLogin: false,
        providers: [
          {
            id: GoogleLoginProvider.PROVIDER_ID,
            provider: new GoogleLoginProvider(
              '408077216324-u368r57tn13rdju7832l1nj7fs7c4mdf.apps.googleusercontent.com'
            )
          }
        ],
        onError: (err) => {
        }
      } as SocialAuthServiceConfig
    }
  ],
  bootstrap: [AppComponent]
})
export class AppModule { }
