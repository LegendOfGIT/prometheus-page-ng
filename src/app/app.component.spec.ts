import { TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';

import { AppComponent } from './app.component';
import { WishlistItemsApiService } from './service/wishlist-items-api.service';

describe('AppComponent', () => {
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        RouterTestingModule
      ],
      declarations: [
        AppComponent
      ],
      providers: [
        { provide: 'API_BASE', useValue: 'http://base.de' },
        { provide: WishlistItemsApiService, useValue: {} }
      ]
    }).compileComponents();
  });

  it('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.componentInstance;
    expect(app).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

  it(`should have as title 'prometheus-page'`, () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.componentInstance;
    expect(app.title).toEqual('prometheus-page');
    expect(fixture).toMatchSnapshot();
  });

});
