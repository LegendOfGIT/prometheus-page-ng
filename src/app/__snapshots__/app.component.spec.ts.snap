// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should create the app 1`] = `
<app-root
  __ngContext__="0"
  destroyedService$={[Function Subject]}
  title={[Function String]}
  wishlistItemsService={[Function Object]}
>
  <div
    class="app"
  >
    <header
      class="prometheus-header"
    >
      <section
        class="prometheus-header__content"
      >
        <header />
      </section>
    </header>
  </div><div
    class="app-content"
  >
    <router-outlet />
  </div>
</app-root>
`;

exports[`AppComponent should have as title 'prometheus-page' 1`] = `
<app-root
  __ngContext__={[Function Number]}
  destroyedService$={[Function Subject]}
  title={[Function String]}
  wishlistItemsService={[Function Object]}
>
  <div
    class="app"
  >
    <header
      class="prometheus-header"
    >
      <section
        class="prometheus-header__content"
      >
        <header />
      </section>
    </header>
  </div><div
    class="app-content"
  >
    <router-outlet />
  </div>
</app-root>
`;
